### DISCUSSION TITLE
singleton container postgresql

### DISCUSSION THREAD

**Discussion by diogo-soares** [testcontainers/testcontainers-dotnet#924] [Q&A] [answered]
Is it possible to create a postgres container for example and run all my table creation scripts run all my tests with just one container instead of creating multiple database instances unnecessarily ?

**Comment by HofmeisterAn** [comment, marked as answer]
OC, why wouldn't that be possible? The singleton pattern works well with Testcontainers. We also use it in our library. For instance, both the Resource Reaper and the Port Forwarding Container are implemented as singletons.
